src/frontend/contexts/dataContext.js

```diff
@@ -0,0 +1,62 @@
+import React, { createContext , useContext , useReducer, useState } from 'react';
+import { handleAllFilters , initialFilters , productInCartWishlistChecker } from '../services/filtersService';
+import axios from 'axios';
+import { useUser } from './userContext';
+
+const DataContext = createContext();
+
+const DataProvider = ({ children }) => {
+    const [ products , setProducts] = useState([]);
+    const { userState }  = useUser();
+    axios.get('/api/products')
```

Comment:
Try using Async await for API calls

Reply:
sure Rama , will change these to async await calls